cmd/crossplane/core/core.go

```diff
@@ -58,6 +59,8 @@ type startCommand struct {
 	LeaderElection bool          `short:"l" help:"Use leader election for the controller manager." default:"false" env:"LEADER_ELECTION"`
 	Registry       string        `short:"r" help:"Default registry used to fetch packages when not specified in tag." default:"${default_registry}" env:"REGISTRY"`
 	Sync           time.Duration `short:"s" help:"Controller manager sync period duration such as 300ms, 1.5h or 2h45m" default:"1h"`
+
+	EnableAlphaCompositionRevisions bool `help:"Enable alpha support for CompositionRevisions."`
```

Comment:
Yeah, I thought about something like this too. Personally I feel that the UX of distinct flags is better, but I agree that makes them tricky to deprecate. One thing we could do would be to keep the flags around for a few releases even after the feature has been promoted to beta.

Reply:
> One thing we could do would be to keep the flags around for a few releases even after the feature has been promoted to beta.

Just noting here that I believe we _must_ keep them for beta, as that would mean they are "on by default", but can still be turned off. I also like the idea of having distinct flags because a configuration value silently not modifying behavior is negative attribute IMO (i.e. I want to break if I tell a service to behave in a way it does not understand) 👍🏻 

@negz I do wonder if it would make sense to add `default:"false"` to the flag so it will show in the help output (I know typically a bool flag will always be `false` if unspecified, but if we follow the k8s "on by default" strategy this bool flag will actually default to `true` in the future). You can also make the help output group all feature flags together using a group (example: https://github.com/upbound/up/blob/8e5965a42eff0285820247abed47cd91f2d26848/cmd/up/uxp/uxp.go#L40)

Reply:
> I do wonder if it would make sense to add default:"false" to the flag so it will show in the help output 

Unfortunately while `default` does appear to actually influence the default value it doesn't change the `--help` output in any way. That said, I do think there's a clean way to handle promoting features to beta. Specifically:

* We don't mention `beta` or `alpha` in the flag name. I've changed this to just `--enable-composition-revisions`.
* We use the `group` as you suggested to make it clear whether a feature is alpha or beta.
* When a feature reaches beta we add `default:"true" negatable:""` to the tags.

This means anyone who is enabling the feature by passing `--enable-composition-revisions` at alpha will not be affected when the feature hits beta. Anyone who wants to disable the feature when it hits beta can pass `--no-enable-composition-revisions`.